### PR TITLE
test: More iceberg tests.

### PR DESCRIPTION
**Pull request by liurenjie1024**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Clsoses #11650

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: ci/spark-script/spark-connect-server.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+set -ex
+
+ICEBERG_VERSION=1.3.1
+SPARK_VERSION=3.4.1
+
+PACKAGES="org.apache.iceberg:iceberg-spark-runtime-3.3_2.12:$ICEBERG_VERSION,org.apache.hadoop:hadoop-aws:3.3.2"
```
Comment (wenym1) [inline ci/spark-script/spark-connect-server.sh:6]: The spark version in the package version is not consistent with the spark image?

**Review by wenym1 — APPROVED** [review]
Rest LGTM!

File: e2e_test/iceberg/main.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    tc = unittest.TestCase()
+    print(f"Executing sql: {sql}")
+    global spark
+    df = spark.sql(sql).collect()
+    for row in df:
+        print(row)
+
+    with open(args['default']['result'], newline='') as csv_file:
+        csv_result = csv.reader(csv_file)
+        for (row1, row2) in zip(df, csv_result):
```
Comment (ZENOTME) [inline e2e_test/iceberg/main.py:48]: zip will take the smaller len. So we may need add len check before.

**Review by ZENOTME — APPROVED** [review]
Rest LGTM

**Review by tabVersion — APPROVED** [review]

**Pull request merged**